src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/corporates/CorporateResourceIntTest.kt

```diff
@@ -524,6 +524,77 @@ class CorporateResourceIntTest : IntegrationTestBase() {
     }
   }
 
+  @Nested
+  @DisplayName("DELETE /corporates/{corporateId}")
+  inner class DeleteCorporate {
+    private lateinit var corporate: Corporate
+
+    @BeforeEach
+    fun setUp() {
+      nomisDataBuilder.build {
+        corporate = corporate(corporateName = "Shipley Young Hope") {
+          type("YOTWORKER")
+        }
+      }
+    }
+
+    @AfterEach
+    fun tearDown() {
+      corporateRepository.deleteAll()
+    }
+
+    @Nested
+    inner class Security {
+
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.delete().uri("/corporates/${corporate.id}")
+          .headers(setAuthorisation(roles = listOf()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.delete().uri("/corporates/${corporate.id}")
+          .headers(setAuthorisation(roles = listOf("BANANAS")))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access unauthorised with no auth token`() {
+        webTestClient.delete().uri("/corporates/${corporate.id}")
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+    }
+
+    @Nested
+    inner class Validation {
+      @Test
+      fun `will return 204 if corporate does not exist`() {
+        webTestClient.delete().uri("/corporates/99999")
+          .headers(setAuthorisation(roles = listOf("NOMIS_CONTACTPERSONS")))
+          .exchange()
+          .expectStatus().isNoContent
+      }
+    }
+
+    @Nested
+    inner class HappyPath {
+      @Test
+      fun `will delete the corporate`() {
```

Comment:
would be good to have a test to ensure that delete of a corporate that didn't exist still succeeds with a 204

Reply:
Just above in the Validation section :-)